### DISCUSSION TITLE
nuxt layers

### DISCUSSION THREAD

**Discussion by ilyam2014** [nuxt/nuxt#26131] [Questions]
after configuring components' path prefix in a base app like that
![image](https://github.com/nuxt/nuxt/assets/89526742/96d9b855-1126-4a4a-906f-afc9fa59a191)
they are not imported in an app which extends the base app. Why?

**Comment by AloisSeckar** [comment]
I found out just recently, you need to do the same config in extending app as well. Or at least I didn't find any way how to avoid that.

But so far I am struggling with correct reference for the base layer, if it is not included in the codebase, but it is imported as an npm package, question asked here https://github.com/nuxt/nuxt/discussions/26127

**Reply by ilyam2014** [reply]
sorry, didnt notice your question, i guess i can close this one and we'll keep track of this in your discussion

**Reply by ilyam2014** [reply]
well in my case the path to the base app is something like `node_modules\.c12\github_base_app_foZCwEoamP` which obviously cannot be set in the extending app's config though it also works for me in dev mode (haven't tested in production just yet)

**Reply by AloisSeckar** [reply]
The ideal scenario would be not having to deal with this. The base layer should know the list of components it is bringing in with their actual names. The extending layer should be notified, that it shouldn't try to resolve names path-based, just by reading the base config.  Or, if it is not possible directly, via some simple config prop inside the `extends`.

Unfortunately, I am not familiar enough with the Nuxt internals to see, how complicated this would be. Maybe there are some technical blocks preventing such behavior.

**Reply by ilyam2014** [reply]
you are sponsor so how fast that will be resolved?

**Reply by AloisSeckar** [reply]
well...my "sponzorship" means I contribute $12 monthly to Nuxt project (and another $8 to Evan Yu for Vue itself). That hardly gives me any right to push Nuxt team into doing anything :)
